producers/mysql-producer/src/main/java/com/flipkart/aesop/runtime/producer/avro/MysqlAvroEventManager.java

```diff
@@ -168,6 +194,34 @@ public int createAndAppendEvent(DbChangeEntry changeEntry, DbusEventBufferAppend
 	}
 
 	/**
+	 * This method scans new and old records and calculates changes.
+	 *
+	 * @param oldGenericRecord is a generic rocord representing old-record before change
+	 * @param newGenericRecord is a generic rocord representing new-record after change
+	 * @return HashMap<String, Object> this contains changes in form of <columnName,ColumnValue>
+	 */
+	private Map<String, Object> calculateChange(GenericRecord oldGenericRecord, GenericRecord newGenericRecord)
```

Comment:
Mainly because its doing some static computation and does not have any state to maintain ? 
Also is it better to move this method to some utility file or its ok to be here?